tools/dataset_converters/waymo_gtbin_creator.py

```diff
@@ -0,0 +1,178 @@
+                                or label.num_lidar_points_in_box < 1):
+                            continue
+
+                        new_obj = metrics_pb2.Object()
+                        new_obj.frame_timestamp_micros = frame.timestamp_micros
+                        new_obj.object.CopyFrom(label)
+                        new_obj.context_name = frame.context.name
+                        objs.objects.append(new_obj)
+                prog_bar.update()
+
+            open(self.waymo_bin_file, 'wb').write(objs.SerializeToString())
+            print(f'Saved groudtruth bin file to {self.waymo_bin_file}\n\
+                    It has {len(objs.objects)} objects in {frame_num} frames.')
+
+        return self.waymo_bin_file
+
+    def get_file_names(self):
+        """Get file names of waymo raw data."""
+        if 'path_mapping' in self.file_client_args:
+            for path in self.file_client_args['path_mapping'].keys():
+                if path in self.waymo_tfrecords_dir:
+                    self.waymo_tfrecords_dir = \
+                        self.waymo_tfrecords_dir.replace(
+                            path, self.file_client_args['path_mapping'][path])
+            from petrel_client.client import Client
+            client = Client()
+            contents = client.list(self.waymo_tfrecords_dir)
+            self.waymo_tfrecord_pathnames = list()
+            for content in sorted(list(contents)):
+                if content.endswith('tfrecord'):
+                    self.waymo_tfrecord_pathnames.append(
+                        join(self.waymo_tfrecords_dir, content))
+        else:
+            self.waymo_tfrecord_pathnames = sorted(
+                glob(join(self.waymo_tfrecords_dir, '*.tfrecord')))
+        print(len(self.waymo_tfrecord_pathnames), 'tfrecords found.')
+
+
+def parse_args():
+    from argparse import ArgumentParser
+    parser = ArgumentParser()
+    parser.add_argument(
+        '--ann_file',
+        default='./data/waymo_dev1x/kitti_format/waymo_infos_val.pkl')
+    parser.add_argument(
+        '--data_root', default='./data/waymo_dev1x/waymo_format')
+    parser.add_argument(
+        '--split', default='validation')  # ('training','validation')
+    # parser.add_argument('waymo_bin_file')
+    parser.add_argument('--load_interval', type=int, default=1)
+    parser.add_argument('--for_cam_only_challenge', default=True)
+
+    args = parser.parse_args()
+    return args
+
+
+def main():
+    args = parse_args()
+    # test = './data/waymo_dev1x/waymo_format/gt.bin'
+    creator = gt_bin_creator(args.ann_file, args.data_root, args.split, None,
+                             args.load_interval, args.for_cam_only_challenge)
+    waymo_bin_file = creator.create_whole()
+    waymo_subset_bin_file = creator.create_subset()
+    print(waymo_bin_file, waymo_subset_bin_file)
+    # breakpoint()
+
+
+if __name__ == '__main__':
+    main()
+"""
+Usage:
+python tools/dataset_converters/waymo_gtbin_creator.py \
```

Comment:
Please remember modifying this script after changing the file name to waymo_gt_bin_creator